server/src/main/java/org/elasticsearch/action/downsample/DownsampleConfig.java

```diff
@@ -55,19 +58,23 @@ public class DownsampleConfig implements NamedWriteable, ToXContentObject {
     private static final String NAME = "downsample/action/config";
     public static final String FIXED_INTERVAL = "fixed_interval";
     public static final String TIME_ZONE = "time_zone";
+    public static final String TIMEOUT = "timeout";
     public static final String DEFAULT_TIMEZONE = ZoneId.of("UTC").getId();
+    public static final TimeValue DEFAULT_TIMEOUT = new TimeValue(1, TimeUnit.DAYS);
 
     private static final String timestampField = DataStreamTimestampFieldMapper.DEFAULT_PATH;
     private final DateHistogramInterval fixedInterval;
     private final String timeZone = DEFAULT_TIMEZONE;
     private final String intervalType = FIXED_INTERVAL;
+    private final TimeValue timeout;
```

Comment:
Should this perhaps be a cluster setting instead? I what case would someone want to configure a different timeout for different downsample operations? Or define a timeout at all?

Reply:
As discussed offline I made it as a rest api request param.